src/web/command_control/client/components/HubLayers.ts

```diff
@@ -88,11 +68,38 @@ export class HubLayers {
 
             const hubLayer = this.getHubLayer(hub_id)
             const hubSource = hubLayer.getSource()
-            const hubFeature = this.getHubFeature(hub, this.map, hubSource)
-    
+            let hubFeature: Feature<Point> = hubSource.getFeatureById(hub_id) as Feature<Point>
+            if (hubFeature == null) {
+                hubFeature = new Feature<Point>({
+                    name: String(hub_id),
+                })
+                hubFeature.setId(hub_id)
+                hubSource.addFeature(hubFeature)
+            }
+
+            if (hub?.location !== undefined) {
+                hubFeature.setGeometry(new Point(getMapCoordinate(hub.location, this.map)))
+            }
+
+            hubFeature.setId(hub.hub_id)
+            hubFeature.setStyle(Styles.hubMarker)
+            hubFeature.setProperties({
+                'type': 'hub',
+                'hub': hub
+            })
+
             const selected = selectedHubOrBot != null && selectedHubOrBot.type == "hub" && selectedHubOrBot.id == hub_id
             hubFeature.set('selected', selected);
+
+            hubLayer.changed();
+
         } // end foreach hub
-    }
-}
 
+        // Remove hub layers for hub_ids that have disappeared
+        const defunctHubIds = Object.keys(this.layers).filter((hub_id) => {return !(String(hub_id) in hubs)})
```

Comment:
I think we can make this cleaner with by removing the brackets and the return keyword
`const defunctHubIds = Object.keys(this.layers).filter((hub_id) =>  !(String(hub_id) in hubs))`